app/src/main/java/fr/free/nrw/commons/notification/NotificationActivity.java

```diff
@@ -154,6 +196,32 @@ private void addNotifications() {
         }
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_notifications, menu);
+        notificationmenuitem = menu.findItem(R.id.archived);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle item selection
+        switch (item.getItemId()) {
+            case R.id.archived:
+                if (item.getTitle().equals(getString(R.string.menu_option_archived))) {
```

Comment:
call 

```
NotificationActivity.startYourself(this, "read");
```

This is how the Wikipedia app does it. It opens a new activity.

Reply:
Does that mean creating multiple instace of the same class?

Reply:
Yes having it open a new activity. It's fine to have multiple instances of an activity class as it would eliminate the need for state management. If you see the Wikipedia app, they do it in a similar fashion. They open a new activity on clicking view archived and then add the original activity to back stack

Reply:
In startyourself() method in notification activity should i remove these flags ?

>intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT | Intent.FLAG_ACTIVITY_SINGLE_TOP);